### ISSUE TITLE
code

### ISSUE THREAD

**Issue by wjczf123** [mark-xhchen/Conditional-ECPE#2]
I have one question:
In function load_data, what do the meanings of cau, con, and con_doc_len do? I am very confused.
I am very sorry to disturb you.

**Comment by mark-xhchen** [comment]
cau is to store the index of the cause clauses, and con is to store the index of the context clauses. The emo, cau and con variables can be used when we need to separately extract the feature vectors of the emotion clauses, cause clauses, and the context clauses, respectively. As for the con_doc_len, I think it is a variable in a previous version and I forgot to delete it in the current version. You can just neglect it. 

Also, I received another email from you that you found that PAM is not working, but I cannot find the response issue so I haven't got a chance to reply. I think mostly it is because of random seed. I am extending the dataset and do some recoding recently. I will let you know if we update the dataset and the codes. Sorry for any inconvenience caused. Nevertheless, the current dataset should be fine and feel free to use it.

**Comment by wjczf123** [comment]
@mark-xhchen Thank you for your reply. It seems that emo is also to store the index of the emotion clauses. But the shape is different. emo is 9376 *1 * 75 and cause is emo is 9376 * 3 * 75.
 
I also think it is because of random seed. So I delete this problem. Thank you a lot.

**Comment by mark-xhchen** [comment]
Yes, emo is also to store the index. In the current dataset, for each document, there will only be one emotion clause, but there can be at most 3 cause clauses. So their shapes are different, and due to the requirement of fixed shape in tensorflow, the shape of cau is  set to [None, 3, 75].

**Comment by wjczf123** [comment]
OK. Thank you a lot.

**Comment by wjczf123** [comment]
@mark-xhchen If the document has more cause clauses (i.e., more than one cause clause), how do we calculate the final result? 
And I hypothesis P(y_i)=(0.65,0.3), calculate by (9). Then the final probability distribution of prediction is to P(y_i) = (0.65/0.9,0.3/0.9)?Then we use cross entropy to get the loss?

**Comment by mark-xhchen** [comment]
The calculation procedure will not be affected by the number of cause clauses. No matter how many cause clauses a document has, the input for achieving p(y^o_i) and p(y^c_i) will always have the size [4, hidden_size], where 4 stands for the concatenation of 1 emotion clause and 3 cause clauses (if less than 3, value 0 will be padded to fill in the matrix), and hidden_size is the size of the feature vector.  

As for your example, I am a bit confused. Theoretically, the sum of the final P(y_i) should be 1, meaning your example should be (0.65, 0.35). If we want to make sure that the sum is 1, a sum-to-1 normalization can be conducted. And then we use the cross-entropy to get the loss.

**Comment by wjczf123** [comment]
@mark-xhchen Thank you for your reply. I know the first question.
For the second question, I thought that the sum of P(y_i) might not be 1 (through Equation (9)). Then I thought there may be normalization operation. Now I found the sum of P(y_i) is all 1. I now all details. Thank you a lot.
